### PR TITLE
Replace instances of reactor pumping with get_success.

### PR DESCRIPTION
**Pull request by anoadragon453**
Calls `self.get_success` on all deferred methods instead of abusing `self.pump()`. This has the benefit of working with coroutines, as well as checking that method execution completed successfully.

There are also a few small cleanups that I made in the process.

### CODE REVIEW

File: tests/storage/test_monthly_active_users.py [hunk 1]
```diff
@@ -61,35 +61,41 @@ def test_initialise_reserved_users(self):
...
-        self.assertEquals(active_count, user_num)
+        self.assertEqual(active_count, user_num)
 
         # Test each of the registered users is marked as active
-        timestamp = self.store.user_last_seen_monthly_active(user1)
-        self.assertTrue(self.get_success(timestamp))
-        timestamp = self.store.user_last_seen_monthly_active(user2)
-        self.assertTrue(self.get_success(timestamp))
+        timestamp = self.get_success(self.store.user_last_seen_monthly_active(user1))
+        self.assertGreater(timestamp, 0)
```
Comment (clokep) [inline tests/storage/test_monthly_active_users.py:96]: Why this change?

Comment (anoadragon453) [reply tests/storage/test_monthly_active_users.py:96]: If `timestamp` is `-1`, then this test would succeed, which doesn't make sense for a timestamp.

File: tests/storage/test_monthly_active_users.py [hunk 2]
```diff
@@ -194,26 +204,37 @@ def test_reap_monthly_active_users_reserved_users(self):
...
             "initialise", self.store._initialise_reserved_users, threepids
         )
-        count = self.store.get_monthly_active_count()
-        self.assertTrue(self.get_success(count), initial_users)
+        self.get_success(d)
 
-        users = self.store.get_registered_reserved_users()
-        self.assertEquals(len(self.get_success(users)), reserved_user_number)
+        count = self.get_success(self.store.get_monthly_active_count())
+        self.assertTrue(count, initial_users)
```
[the remark this answers is not in the document]
Comment (anoadragon453) [reply tests/storage/test_monthly_active_users.py:213]: Oop, yep, most certainly is.

File: changelog.d/7619.misc [hunk 3]
```diff
@@ -0,0 +1 @@
+Check that called deferred methods succeed and general cleanup of `MonthlyActiveUsersTestCase` and `TestMauLimit`.
```
Comment (clokep) [inline changelog.d/7619.misc:1]: "deferred methods" is confusing to me, we're checking that each `Deferred` has completed. I'm not sure it even matters that they are `Deferreds` though, so maybe:

```suggestion
Check that all asynchronous tasks succeed and general cleanup of `MonthlyActiveUsersTestCase` and `TestMauLimit`.
```
